radicle-cli/src/commands/node/control.rs

```diff
@@ -0,0 +1,54 @@
+use radicle::node::{Address, Handle as _, NodeId};
+use radicle::Node;
+
+use crate::terminal as term;
+
+pub fn start() -> anyhow::Result<()> {
+    todo!()
+}
+
+pub fn stop(node: Node) -> anyhow::Result<()> {
+    let spinner = term::spinner("Stopping the node...");
+    if let Err(err) = node.shutdown() {
+        spinner.finish();
+        term::warning(&format!("Error occurred while shutting down node: {err}"));
+    } else {
+        spinner.finish();
+        term::success!("Successfully shut down the node")
+    }
+    Ok(())
+}
+
+pub fn connect(node: &mut Node, nid: NodeId, addr: Address) -> anyhow::Result<()> {
+    let spinner = term::spinner(format!(
+        "Connecting to {}@{addr}...",
+        term::format::node(&nid)
+    ));
+    if let Err(err) = node.connect(nid, addr.clone()) {
+        spinner.finish();
+        term::fail(
+            &format!(
+                "Failed to connect to {}@{}",
+                term::format::node(&nid),
+                term::format::secondary(addr)
+            ),
+            &err.into(),
+        )
+    } else {
+        spinner.finish();
+        term::success!(
+            "Successfully connected to {}@{}",
+            term::format::node(&nid),
+            term::format::secondary(addr)
+        )
+    }
```

Comment:
Same comments as above, the spinner should be used to display success/failure messages instead of a separate line, because when you call `finish()` it will add a checkmark to the spinner already.

Reply:
So when I have this code:
```rust
        spinner.message(format!(
            "Successfully connected to {}@{}",
            term::format::node(&nid),
            term::format::secondary(addr)
        ));
        spinner.finish()
 ```
 It only displays:
 ```
 ✓ Successfully connected to z6Mkt67…v4N1tRk@0.0.0.0:3679
 ```
 So it's not showing the:
 ```
 ✓ Connecting to z6Mkt67…v4N1tRk@0.0.0.0:3679...
 ```
 Is this a bug?
 
 The `Spinner` API has me a bit confused. What's the difference between `failed` and `error`? And why does `warn` not take any progress message?